src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.cs

```diff
@@ -151,7 +151,7 @@ private void ProduceTags(TaggerContext<TTag> context, DocumentSnapshotSpan spanT
             buffer?.Properties.TryGetProperty(PredefinedPreviewTaggerKeys.SuppressDiagnosticsSpansKey, out suppressedDiagnosticsSpans);
 
             var buckets = _diagnosticService.GetDiagnosticBuckets(
-                workspace, document.Project.Id, document.Id, context.CancellationToken);
+                workspace, document.Project.Id, document.Id, forPullDiagnostics: false, context.CancellationToken);
```

Comment:
an intermediary concept this PR introduces is that the two major clients of the diagnostics subsystem now pass in what behavior they want when calling into the DiagnosticService singleton.  

i.e. the normal push features call in saying `forPullDiagnostics: false` and the lsp system calls in saying `forPullDiagnostics: true`.  Internally we see which mode we're actually in and we either pass along the diagnostics, or  pass along empty.  in essence only one side will work, and the other will see absolutely no diagnostics ever.

This is not pretty, but it was:

1. the cleanest way i could find to keep both systems around
2. required minimal and mechanical changes to both systems.
3. ensured that every entrypoint was guarded so diagnostics wouldn't accidentally sneak into the wrong client.

Once we are entirely on pull, all the push components can go away (as can tehse booleans).